brother_ql/models.py

```diff
@@ -37,34 +37,159 @@ class Model(object):
+    series_code = attrib(type=int, default=0xFFFF)
+    model_code = attrib(type=int, default=0xFFFF)
+    product_id = attrib(type=int, default=0xFFFF)
 
     @property
     def name(self):
         return self.identifier
 
 ALL_MODELS = [
-  Model('QL-500',   (295, 11811), compression=False, mode_setting=False, expanded_mode=False, cutting=False),
-  Model('QL-550',   (295, 11811), compression=False, mode_setting=False),
-  Model('QL-560',   (295, 11811), compression=False, mode_setting=False),
-  Model('QL-570',   (150, 11811), compression=False, mode_setting=False),
-  Model('QL-580N',  (150, 11811)),
-  Model('QL-600',   (150, 11811)),
-  Model('QL-650TD', (295, 11811)),
-  Model('QL-700',   (150, 11811), compression=False, mode_setting=False),
-  Model('QL-710W',  (150, 11811)),
-  Model('QL-720NW', (150, 11811)),
-  Model('QL-800',   (150, 11811), two_color=True, compression=False, num_invalidate_bytes=400),
-  Model('QL-810W',  (150, 11811), two_color=True, num_invalidate_bytes=400),
-  Model('QL-820NWB',(150, 11811), two_color=True, num_invalidate_bytes=400),
-  Model('QL-1050',  (295, 35433), number_bytes_per_row=162, additional_offset_r=44),
-  Model('QL-1060N', (295, 35433), number_bytes_per_row=162, additional_offset_r=44),
-  Model('QL-1100',  (301, 35434), number_bytes_per_row=162, additional_offset_r=44),
-  Model('QL-1110NWB',(301, 35434), number_bytes_per_row=162, additional_offset_r=44),
-  Model('QL-1115NWB',(301, 35434), number_bytes_per_row=162, additional_offset_r=44),
-  Model('PT-E550W',  (31, 14172), number_bytes_per_row=16),
-  Model('PT-P750W',  (31, 14172), number_bytes_per_row=16),
-  Model('PT-P900W',  (57, 28346), number_bytes_per_row=70),
-  Model('PT-P950NW',  (57, 28346), number_bytes_per_row=70),
+    Model(
+        identifier="QL-500",
+        min_max_length_dots=(295, 11811),
+        compression=False,
+        mode_setting=False,
+        expanded_mode=False,
+        cutting=False,
+        series_code=0x30,
+        model_code=0x4F,
+        product_id=0x2015,
+    ),
+    Model(
+        identifier="QL-550",
+        min_max_length_dots=(295, 11811),
+        compression=False,
+        mode_setting=False,
+        series_code=0x30,
+        model_code=0x4F,
+        product_id=0x2016,
+    ),
+    Model(
+        identifier="QL-560",
+        min_max_length_dots=(295, 11811),
+        compression=False,
+        mode_setting=False,
+        series_code=0x34,
+        model_code=0x31,
+        product_id=0x2027,
+    ),
+    Model(
+        identifier="QL-570",
+        min_max_length_dots=(150, 11811),
+        compression=False,
+        mode_setting=False,
+        series_code=0x34,
+        model_code=0x32,
+        product_id=0x2028,
+    ),
+    Model(
+        identifier="QL-580N",
+        min_max_length_dots=(150, 11811),
+        series_code=0x34,
+        model_code=0x33,
+        product_id=0x2029,
+    ),
+    Model(identifier="QL-600", min_max_length_dots=(150, 11811)),
```

Comment:
Q (non-blocking): Are you getting the series codes from the protocol description? If so - can we add the codes for the QL-600 too?

Reply:
I only added the hardware details that I know of, and the raster command manuals I can find on the web don't have all the identifiers listed. In case of the PT series they are not documented at all, I got the values for PT-P700 and PT-P750W from a config file shipped with Brother's printer configuration tool. The rest will likely have to be crowd-sourced from people who own these label printers.

Reply:
Update: I've found more manuals here, I can fill in the rest if I can find them in these manuals. https://support.brother.com/g/s/es/dev/en/command/reference/index.html?c=eu_ot&lang=en&comple=on&redirect=on

Reply:
Great, thank you!